### ISSUE TITLE
hn_cEvents_build_handler sometimes return an unusable handler

### ISSUE THREAD

**Issue by Burgestrand** [Burgestrand/Hallon#18]
Given this file in the path `spec/bug.rb`:
    $: << File.expand_path('../../lib', **FILE**)
    require 'hallon'
    require_relative 'support/config'

```
session = Hallon::Session.instance(Hallon::APPKEY)
puts "Handler: #{session.handler.inspect}"
```

This will segfault on line #6 when inspecting the handler. I do not yet know why. Here’s some more info:
- [Handler is made and set at session.c:105](https://github.com/Burgestrand/Hallon/blob/6499861e075b04479503c39388ecaaed568991dc/ext/hallon/session.c#L105)
- [Session#handler returns whatever the handler is set to](https://github.com/Burgestrand/Hallon/blob/6499861e075b04479503c39388ecaaed568991dc/ext/hallon/session.c#L268)
- [Macro to build handler is in common.h](https://github.com/Burgestrand/Hallon/blob/6499861e075b04479503c39388ecaaed568991dc/ext/hallon/common.h#L41)

**Comment by Burgestrand** [comment]
Alright, by the looks if it the call to `sp_session_create` is what is making the handler unusable. Removing [the call to it](https://github.com/Burgestrand/Hallon/blob/b9a8752948df0422f6c4b88ce4026bbdc2191f7c/ext/hallon/session.c#L134) won’t cause the error.

Now, how do I fix that…? Hmm!

**Comment by Burgestrand** [comment]
Heh. Forgot to allocate a pointer in cSession_s_alloc :d fix coming up

**Comment by Burgestrand** [comment]
Properly allocate hn_spotify_data_t (Closed by 05f8bb3240550d04243afd2e3c985a80e85734a9)
